kernel/constr.ml

```diff
@@ -1418,8 +1417,6 @@ and sh_invert = function
 
 and sh_rec t =
   let (y, h) = hash_term t in
-  (* [h] must be positive. *)
-  let h = h land 0x3FFFFFFF in
```

Comment:
Are we sure we do not need the hash to be nonnegative? Even the OCaml runtime has special code to make sure that hashes are always nonnegative:
```c
  /* Fold result to the range [0, 2^30-1] so that it is a nonnegative
     OCaml integer both on 32 and 64-bit platforms. */
  return Val_int(h & 0x3FFFFFFFU);
```

Reply:
AFAIK nothing relies on that in the Coq codebase. I assume that the OCaml code does that simply as a best-effort to emulate compatibility between 64 and 32 bit architectures.